### PR TITLE
feature(sidecar): finish some part of the sidecar

### PR DESCRIPTION
**Pull request by shejialuo**
The implementation of the sidecar need to use the webhook functionality provided by k8s. And we need admission controller to finish the work. However, this commit only implments the basic idea.

TODO:

1. Web server
2. Admission controller

### CODE REVIEW

**Review by North-C — APPROVED** [review]
lgtm!!!!

File: util/types/pod.hpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    return "pod";
+  }
+  /**
+   * @brief Generate a random string for pod
+   *
+   * @return std::string generated pad name
+   */
+  std::string getGenerateName() {
+    // TODO: this part is hardcoded. Need to revise later
+    return "pod";
```
Comment (pengliz) [inline util/types/pod.hpp:30]: 与21行类似有魔法数字嫌疑 建议定义常量pod 再去返回

**Review by pengliz — APPROVED** [review]

**Comment by likeran2234** [comment]
well down

**Comment by ChongyangZ** [comment]
good job

**Pull request merged**